### ISSUE TITLE
Win: pyzo does not open file from explorer

### ISSUE THREAD

**Issue by Borim7** [pyzo/pyzo#521]
When associate *.py files with pyzo. Double clicking the file opens pyzo, but only the files from the last session are visible. Not the clicked file.
The same counts when starting pyzo from the command line like:
> pyzo D:\code\rename.py

Tested with pyzo 4.4.2 on Win7 x64

**Comment by almarklein** [comment]
mmm I cannot reproduce this. It works fine here ...  What if you do:
```
pyzo open D:\code\rename.py
```

**Comment by Borim7** [comment]
It is the same, no changes. What additional information can I provide to help you reproduce the bug?

**Comment by almarklein** [comment]
Just to be sure, `D:\code\rename.py` is the actual file that you try to open, and it does exist? E.g. Spaces in paths often cause problems.

**Comment by shamashii** [comment]
Same here with sometimes hard reproducible case, now I'm using some context menu manager adding pyzo to my right clickmenu, same function of "sendto", may help.

**Comment by Borim7** [comment]
Yes this is the file I use to test and it exists. When opened within of pyzo the file is displayed correctly. By the way the bug is triggered for every *.py file.

**Comment by almarklein** [comment]
@shamashii  What OS are you on? I'm on win10 myself, and it also works  fine for me with the latest release.

I don't have time to look into it right now. If anyone is, a good start could be running Pyzo from source to see if that works. Looking further, check [commandlin.py](https://github.com/pyzo/pyzo/blob/master/pyzo/core/commandline.py). E.g. validate that the command server is running.

**Comment by Borim7** [comment]
Testing from source have the same issues, but return some additional info:
```
Could not process command line args:
'NoneType' object has no attribute 'rstrip'
Our command server is *not* running
loading qt German: failed
loading pyzo German: ok
Could not process command line args:
'NoneType' object has no attribute 'rstrip'
```

Lets see where this brings us...

**Comment by Borim7** [comment]
The problem is the command server. For some reason the socket can not receive any data, instead the exception `ConnectionAbortedError: [WinError 10053] An established connection was aborted by the software in your host machine`

pyzo/yoton/core.py, line 164

My first thought was that my firewall interfere with python, but shutting it down does not change anything. So it seems some other problem.

**Comment by almarklein** [comment]
Could you try increasing the timeout of the `do_request()` call in `handle_cmd_args()` in `commandline.py` (what is now 0.4)?

**Comment by Borim7** [comment]
Altering the timeout does not change anything. Also 4 seconds did not help.

The problematic code is not in the receive path, but in `send_all` in `pyzo/yoton/core.py`. The shutdown after sending part, interrupt the transmission or processing of the request. So the server does not see the request or receive only an empty request.
After disable the close after send option by altering the `do_request` function in `pyzo\yoton\clientserver.py` to
````py
    # Send request
    send_all(s, request, False)
````
Pyzo process the command line and open the passed files.

I do not know if this change break anything else, or if closing the port after sending is used unintentionally?

**Comment by almarklein** [comment]
From the docstring in `send_all()` it seems that closing the socket is required in some cases. I am assuming that calling `s.shutdown()` will first flush any pending-to-be-send data, and googling does not seem to suggest otherwise... odd

Anyway, if we can fix this by exposing the `shutdown` param in `do_request()` and then setting it to `False` in `clientserver.py`, than I'd be fine with that.

**Comment by bdieterm** [comment]
Pyzo and operating systems have changed a lot during the last 8 years.
If there is still such a problem, please create a new issue or comment here.
